src/ldp/operations/HeadOperationHandler.ts

```diff
@@ -0,0 +1,37 @@
+import { Readable } from 'stream';
+import type { ResourceStore } from '../../storage/ResourceStore';
+import { UnsupportedHttpError } from '../../util/errors/UnsupportedHttpError';
+import type { Operation } from './Operation';
+import { OperationHandler } from './OperationHandler';
+import type { ResponseDescription } from './ResponseDescription';
+
+/**
+ * Handles HEAD {@link Operation}s.
+ * Calls the getRepresentation function from a {@link ResourceStore}.
+ */
+export class HeadOperationHandler extends OperationHandler {
+  private readonly store: ResourceStore;
+
+  public constructor(store: ResourceStore) {
+    super();
+    this.store = store;
+  }
+
+  public async canHandle(input: Operation): Promise<void> {
+    if (input.method !== 'HEAD') {
+      throw new UnsupportedHttpError('This handler only supports HEAD operations.');
+    }
+  }
+
+  public async handle(input: Operation): Promise<ResponseDescription> {
+    const body = await this.store.getRepresentation(input.target, input.preferences);
+
+    // Close the Readable as we will not return it.
+    process.nextTick((): any => body.data.destroy());
```

Comment:
Not sure why the `nextTick` call is necessary and `body.data.destroy()` can't be called directly?

Reply:
If we do, I get an `Error [ERR_STREAM_DESTROYED]: Cannot call write after a stream was destroyed` error. 
The stream is destroyed too early, I got my inspiration [here](https://stackoverflow.com/a/53878933).

Reply:
Probably will be easier to test then using the Promise solution that's also suggested there.

Reply:
Then we have an uncovered .catch branch that will never be taken, as it is an empty Promise, but is required by the linter.
I'm not sure how I can solve that

Reply:
The `handle` function is supposed to return a Promise anyway, so you could chain them and return that.